### PR TITLE
Feature 390/tarrifs calculation

### PR DESCRIPTION
**Pull request by yevhendevbox**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by Naira21 — DISMISSED** [review]

**Review by anatoliy-stepanchenko — DISMISSED** [review]

**Review by anatoliy-stepanchenko — DISMISSED** [review]

**Review by Naira21 — DISMISSED** [review]

File: src/finance/components/TariffsCalculation.vue [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+            </div>
+            <div class="input_field area-label">
+              <h4>Загальна площа:</h4>
+              <Chip :label="`${area}, м²`" icon="pi pi-pencil" style="max-width: 120px" />
+            </div>
+          </div>
+        </form>
+      </div>
+      <div class="tariffs-calculator--right-col">
+        <form class="tariffs-calculator--service-form">
```
Comment (aleonovdementiev) [inline src/finance/components/TariffsCalculation.vue]: Do you need this "form" element if you have "service item" component?

File: src/finance/components/TariffsCalculation.vue [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+      return store.getters[`${StoreModuleEnum.cooperationStore}/${CooperationGettersEnum.getSelectedCooperationId}`];
+    });
+
+    const setHouses = async () => {
+      houses.value = [
+        { adress: 'Dnipro, Kirova str. 102', houseArea: 250, houseId: 1 },
+        { adress: 'Dnipro, Robocha str. 31', houseArea: 420, houseId: 2 },
+        { adress: 'Dnipro, Bohdana Hmelnitskogo str. 12', houseArea: 350, houseId: 3 },
+        { adress: 'Dnipro, Peremogi str. 23', houseArea: 270, houseId: 4 },
+      ];
```
Comment (aleonovdementiev) [inline src/finance/components/TariffsCalculation.vue]: Why do you need this array here?

**Review by anatoliy-stepanchenko — DISMISSED** [review]

**Review by Naira21 — DISMISSED** [review]

File: src/finance/components/TariffsCalculation.vue [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+                        <Button
+                          icon="pi pi-pencil"
+                          class="p-button-rounded p-button-warning p-button-text"
+                          @click="toggleServiceEdit(service)"
+                        />
+                        <Button
+                          icon="pi pi-times"
+                          class="p-button-rounded p-button-danger p-button-text"
+                          @click="handleServiceDelete(service)"
+                        />
```
Comment (aleonovdementiev) [inline src/finance/components/TariffsCalculation.vue]: Shouldn't this buttons be in a service component??

File: src/finance/components/TariffsCalculation.vue [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+    const toggleServiceEdit = (serviceToEdit: TariffService): void => {
+      const indexToEdit = tariffData.services.indexOf(serviceToEdit);
+      tariffData.services[indexToEdit].editState = !tariffData.services[indexToEdit].editState;
+    };
+
+    const updateService = ({ editedService, idx }: { editedService: TariffService; idx: number }): void => {
+      tariffData.services[idx].serviceTitle = editedService.serviceTitle;
+      tariffData.services[idx].servicePrice = editedService.servicePrice;
+      tariffData.services[idx].editState = !tariffData.services[idx].editState;
+    };
```
Comment (aleonovdementiev) [inline src/finance/components/TariffsCalculation.vue]: All this logic should be inside the service component

File: src/finance/components/ServiceForm.vue [hunk 5]
```diff
@@ -163,13 +194,16 @@ export default defineComponent({
...
-.service_title,
-.service_price {
-  flex-grow: 1;
-  width: 48%;
+.service-form__wrapper {
+  display: flex;
+  width: 100%;
+  gap: 1em;
+  flex-wrap: wrap;
+  align-items: center;
```
Comment (anatoliy-stepanchenko) [inline src/finance/components/ServiceForm.vue]: It will look better without this style.

File: src/finance/components/TariffsCalculation.vue [hunk 6]
```diff
@@ -72,7 +72,7 @@
                   <service-form
```
Comment (anatoliy-stepanchenko) [inline src/finance/components/TariffsCalculation.vue:72]: maybe it's better not to post a comment here? I mean line 69.

**Review by anatoliy-stepanchenko — DISMISSED** [review]

File: src/finance/components/TariffsCalculation.vue [hunk 7]
```diff
@@ -0,0 +1,422 @@
...
+    });
+
+    const setHouses = async () => {
+      await store.dispatch(`${StoreModuleEnum.housesStore}/${HousesActionsEnum.SET_HOUSES}`, cooperationId.value);
+      const housesList = await store.getters[`${StoreModuleEnum.housesStore}/${HousesGettersEnum.getHousesData}`];
+      houses.value = housesList.reduce((acc: any, house: HouseModel) => {
+        return (acc = [
+          ...acc,
+          {
+            adress: `${house.address.city}, ${house.address.street}, ${house.address.houseNumber},
```
Comment (aleonovdementiev) [inline src/finance/components/TariffsCalculation.vue]: address?

File: src/finance/components/TariffsCalculation.vue [hunk 8]
```diff
@@ -0,0 +1,422 @@
...
+      submitted.value = false;
+      tariffData.house = {} as SelectedHouse;
+      tariffData.title = '';
+      tariffData.comment = '';
+      tariffData.services = [];
+      localStorage.removeItem('current-tariff');
+    };
+
+    const addService = (newService: TariffService): void => {
+      tariffData.services = [...tariffData.services, newService];
```
Comment (aleonovdementiev) [inline src/finance/components/TariffsCalculation.vue]: Maybe 'tariffData.services.push'?

**Review by aleonovdementiev — DISMISSED** [review]

**Review by anatoliy-stepanchenko — DISMISSED** [review]

**Review by Naira21 — DISMISSED** [review]

**Review by Naira21 — APPROVED** [review]

**Review by anatoliy-stepanchenko — APPROVED** [review]

**Pull request merged**